cli/src/config.rs

```diff
@@ -583,6 +584,58 @@ fn read_config_path(config_path: &Path) -> Result<config::Config, config::Config
         .build()
 }
 
+pub fn remove_config_value_from_file(
+    key: &ConfigNamePathBuf,
+    path: &Path,
+) -> Result<(), CommandError> {
+    // Read config
+    let config_toml = std::fs::read_to_string(path).or_else(|err| {
+        match err.kind() {
+            // If config doesn't exist yet, read as empty and we'll write one.
+            std::io::ErrorKind::NotFound => Ok("".to_string()),
+            _ => Err(user_error_with_message(
+                format!("Failed to read file {path}", path = path.display()),
+                err,
+            )),
+        }
+    })?;
+    let mut doc: toml_edit::Document = config_toml.parse().map_err(|err| {
+        user_error_with_message(
+            format!("Failed to parse file {path}", path = path.display()),
+            err,
+        )
+    })?;
```

Comment:
nit: Perhaps, we can extract read/write helper functions or a function that takes callback `update_doc: impl FnOnce(&mut Document) -> Result<..>`.

Reply:
I really liked the idea of an `update_doc` callback, but it turned out to be a little bumpy. `write_config_value_to_file()` takes a `value` argument, whereas `remove_config_value_from_file()` does not. I tried making `value` optional, but that made the interface awkward.  

That's when I decided to go for `read_config()` and `write_config()` helpers, despite not being too happy about it. 

Since I do not consider myself an advanced Rustacean, feel free to give it a shot yourself; I might be missing something.

Reply:
I just realized I baked the refactoring and the feature into one change. I will split these into two separate ones.